src/map/lua/lua_baseentity.cpp

```diff
@@ -4106,15 +4106,16 @@ void CLuaBaseEntity::confirmTrade()
 
 /************************************************************************
  *  Function: tradeComplete()
- *  Purpose : Completes trade and removes all items in trade container
+ *  Purpose : Completes trade and removes all items in trade container (unless false is passed as parameter)
  *  Example : player:tradeComplete()
  ************************************************************************/
 
-void CLuaBaseEntity::tradeComplete()
+void CLuaBaseEntity::tradeComplete(sol::object const& shouldTakeItems)
```

Comment:
Hmmm can this just be done through changing these interactions to like:
```
trade:hasItemQty
trade:getItemCount() == 1
```
? I genuinely don't know, but I do know these functions exist

Reply:
Sure, but that defeats the purpose of the utility function.  It also would cause a bigger mess if the user was ever required to provide more than 1 item.  This solution is most consistent with the other trade interactions.

Reply:
I don't understand why this needs to be done. Simply not writing player:tradeComplete() prevents the trade from taking the items.

Reply:
The utility function for `npcUtil.tradeHas()` confirms all the items at the end of it, which leaves them in an unusable state (can't be dropped, etc. until zoning) which makes me think we should not be using it here? (As opposed to confirming then unconfirming the items)

Reply:
Is this ready to go? Or are yall asking for changes?